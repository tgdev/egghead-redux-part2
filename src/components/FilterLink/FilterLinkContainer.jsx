import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { setVisibilityFilter } from 'actions/visibility-filter.actions';

import FilterLink from './FilterLink';

const mapStateToProps = (state, ownProps) => {
  return {
    active: ownProps.filter === state.visibilityFilter
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    handleFilter: () => {
      dispatch(setVisibilityFilter(ownProps.filter));
    }
  };
};

const FilterLinkContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(FilterLink);

FilterLinkContainer.propTypes = {
  filter: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired
};

export default FilterLinkContainer;
