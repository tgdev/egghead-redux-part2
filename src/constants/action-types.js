export const ACTION_ADD_TODO = 'ADD_TODO';
export const ACTION_TOGGLE_TODO = 'TOGGLE_TODO';

export const ACTION_SET_VISIBILITY_FILTER = 'SET_VISIBILITY_FILTER';
