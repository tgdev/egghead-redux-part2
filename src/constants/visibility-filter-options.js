export const FILTER_OPTION_ALL = 'SHOW_ALL';
export const FILTER_OPTION_ACTIVE = 'SHOW_ACTIVE';
export const FILTER_OPTION_COMPLETED = 'SHOW_COMPLETED';
