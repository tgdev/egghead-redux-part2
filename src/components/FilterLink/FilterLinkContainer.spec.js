import React from 'react';
import { shallow } from 'enzyme';

import { FILTER_OPTION_ALL } from 'constants/visibility-filter-options';

import appStore from '../../store';

import FilterLinkContainer from './FilterLinkContainer';

describe('FilterLinkContainer component', () => {

  const mockWithContext = () => {
    const mockFilterOption = FILTER_OPTION_ALL;
    const storeContext =  { context: { store: appStore } };

    return shallow(
      <FilterLinkContainer
        filter={mockFilterOption}>
        All
      </FilterLinkContainer>,
      storeContext
    );

  };

  it('renders FilterLinkContainer correctly', () => {
    const wrapper = mockWithContext();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with required children', () => {
    const wrapper = mockWithContext();
    expect(wrapper.children().length).toBe(1);
  });

  it('renders with required filter prop', () => {
    const wrapper = mockWithContext();
    const filterProp = wrapper.instance().props.filter;
    expect(filterProp).toEqual(FILTER_OPTION_ALL);
  });

});
