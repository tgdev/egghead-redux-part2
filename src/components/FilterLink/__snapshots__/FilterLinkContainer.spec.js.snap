// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLinkContainer component renders FilterLinkContainer correctly 1`] = `
<FilterLink
  active={true}
  filter="SHOW_ALL"
  handleFilter={[Function]}
>
  All
</FilterLink>
`;
